huaweicloud/services/dws/data_source_huaweicloud_dws_workload_queues.go

```diff
@@ -0,0 +1,208 @@
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceWorkloadQueuesRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var (
+		cfg     = meta.(*config.Config)
+		region  = cfg.GetRegion(d)
+		httpUrl = "v2/{project_id}/clusters/{cluster_id}/workload/queues"
+		product = "dws"
+	)
+
+	getClient, err := cfg.NewServiceClient(product, region)
+	if err != nil {
+		return diag.Errorf("error creating DWS client: %s", err)
+	}
+
+	getPath := getClient.Endpoint + httpUrl
+	getPath = strings.ReplaceAll(getPath, "{project_id}", getClient.ProjectID)
+	getPath = strings.ReplaceAll(getPath, "{cluster_id}", d.Get("cluster_id").(string))
+
+	// If cluster is logical cluster should add parameter
+	if logicalName, ok := d.GetOk("logical_cluster_name"); ok {
+		getPath += "?logical_cluster_name=" + logicalName.(string)
+	}
+	getOpt := golangsdk.RequestOpts{
+		MoreHeaders:      requestOpts.MoreHeaders,
+		KeepResponseBody: true,
+	}
+
+	getResp, err := getClient.Request("GET", getPath, &getOpt)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	getRespBody, err := utils.FlattenResponse(getResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	queuesListJson := utils.PathSearch("workload_queue_name_list", getRespBody, make([]interface{}, 0))
+	queuesList := queuesListJson.([]interface{})
+
+	uuid, err := uuid.GenerateUUID()
+	if err != nil {
+		return diag.Errorf("unable to generate ID: %s", err)
+	}
+	d.SetId(uuid)
+
+	mErr := multierror.Append(
+		d.Set("region", region),
+		d.Set("queues", filterQueues(queuesList, d, getClient)),
+	)
+
+	return diag.FromErr(mErr.ErrorOrNil())
+}
+
+func filterQueues(all []interface{}, d *schema.ResourceData, client *golangsdk.ServiceClient) []interface{} {
+	rst := make([]interface{}, 0, len(all))
+	for _, v := range all {
+		if param, ok := d.GetOk("name"); ok {
+			if !strings.Contains(v.(string), fmt.Sprint(param)) {
+				continue
+			}
+		}
+
+		rst = append(rst, getQueueDetail(v.(string), d, client))
+	}
+	return rst
+}
+
+func getQueueDetail(queueName string, d *schema.ResourceData, client *golangsdk.ServiceClient) interface{} {
+	httpUrl := "v2/{project_id}/clusters/{cluster_id}/workload/queues/{queue_name}"
+	getPath := client.Endpoint + httpUrl
+	getPath = strings.ReplaceAll(getPath, "{project_id}", client.ProjectID)
+	getPath = strings.ReplaceAll(getPath, "{cluster_id}", d.Get("cluster_id").(string))
+	getPath = strings.ReplaceAll(getPath, "{queue_name}", queueName)
+	// If cluster is logical cluster should add parameter
+	if logicalName, ok := d.GetOk("logical_cluster_name"); ok {
+		getPath += "?logical_cluster_name=" + logicalName.(string)
+	}
+	getOpt := golangsdk.RequestOpts{
+		MoreHeaders:      requestOpts.MoreHeaders,
+		KeepResponseBody: true,
+	}
+
+	getResp, err := client.Request("GET", getPath, &getOpt)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	getRespBody, err := utils.FlattenResponse(getResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
```

Comment:
do not return the error, just record the error message in the terraform log and return nil to continue the loop.
there is unnecessary to stop the queries for the other queue when the someone failure.

Reply:
done